src/trio/_core/_io_kqueue.py

```diff
@@ -98,7 +99,7 @@ def process_events(self, events: EventResult) -> None:
             if isinstance(receiver, _core.Task):
                 _core.reschedule(receiver, outcome.Value(event))
             else:
-                receiver.put_nowait(event)  # TODO: test this line
+                receiver.send_nowait(event)  # TODO: test this line
```

Comment:
it's possible for this to fail if someone calls:

```python
with trio.lowlevel.monitor_kevent() as q:
    q.close()
```

we should either catch the exception here or yield a clone

I'd add it to this PR for you but I just cannot work out how to cover this line - it should be easy but I'm new to kqueue so might be missing something

Reply:
something like this:

```patch
from 9ddb5ecc47dda3a7502851967bb67247a99aefe5 mon sep 17 00:00:00 2001
from: thomas grainger <tagrain@gmail.com>
date: tue, 24 dec 2024 08:24:41 +0000
subject: [patch] make sure closing monitor queues is a noop

---
 src/trio/_core/_io_kqueue.py  | 12 ++++++------
 src/trio/_core/_io_windows.py | 13 +++++++------
 2 files changed, 13 insertions(+), 12 deletions(-)

diff --git a/src/trio/_core/_io_kqueue.py b/src/trio/_core/_io_kqueue.py
index fed4da83..08b5d966 100644
--- a/src/trio/_core/_io_kqueue.py
+++ b/src/trio/_core/_io_kqueue.py
@@ -137,12 +137,12 @@ class kqueueiomanager:
                 "attempt to register multiple listeners for same ident/filter pair",
             )
         send, recv = open_memory_channel[select.kevent](math.inf)
-        self._registered[key] = send
-        try:
-            yield recv
-        finally:
-            send.close()
-            self._registered.pop(key, none)
+        with send, recv:
+            self._registered[key] = send
+            try:
+                yield recv.clone()
+            finally:
+                self._registered.pop(key, none)
 
     @_public
     async def wait_kevent(
diff --git a/src/trio/_core/_io_windows.py b/src/trio/_core/_io_windows.py
index 4676e6c5..824e91f5 100644
--- a/src/trio/_core/_io_windows.py
+++ b/src/trio/_core/_io_windows.py
@@ -1040,9 +1040,10 @@ class windowsiomanager:
 
         key = next(self._completion_key_counter)
         send, recv = open_memory_channel[object](math.inf)
-        self._completion_key_queues[key] = send
-        try:
-            yield (key, recv)
-        finally:
-            send.close()
-            del self._completion_key_queues[key]
+        with send, recv:
+            self._completion_key_queues[key] = send
+            try:
+                yield (key, recv.clone())
+            finally:
+                send.close()
+                del self._completion_key_queues[key]
-- 
2.43.0
```